ouroboros-network/src/Ouroboros/Network/PeerSelection/Churn.hs

```diff
@@ -94,6 +100,27 @@ peerChurnGovernor tracer deadlineChurnInterval bulkChurnInterval psOverallTimeou
                        min 1 (targetNumberOfActivePeers base - 1)
         })
 
+    increaseEstablishedPeers :: ChurnMode -> UseBootstrapPeers -> STM m ()
+    increaseEstablishedPeers mode ubp =  do
+        modifyTVar peerSelectionVar (\targets -> targets {
+          targetNumberOfEstablishedPeers =
+              case (mode, ubp) of
+                   (ChurnModeBulkSync, UseBootstrapPeers _) ->
+                       min (targetNumberOfActivePeers targets + 1) (targetNumberOfEstablishedPeers base)
+                   _  -> targetNumberOfEstablishedPeers base
+        })
+
+    decreaseEstablished :: ChurnMode -> UseBootstrapPeers -> STM m ()
+    decreaseEstablished mode ubp =  do
+        modifyTVar peerSelectionVar (\targets -> targets {
+          targetNumberOfEstablishedPeers =
+              case (mode, ubp) of
+                   (ChurnModeBulkSync, UseBootstrapPeers _) ->
+                       min (targetNumberOfActivePeers targets) (targetNumberOfEstablishedPeers base - 1)
```

Comment:
This means we will remove all non-active established peers if `targetNumberOfActivePeers < targetNumberOfEstablisehdPeers` and if both targets are equal then it will be decreased by one, in which case we will not demote any established peer because all of them are active; it also violates the `sanePeerSelectionTargets` (established becomes smaller than active).

Reply:
When is this mode `targetNumberOfActivePeers = 2` and `targetNumberOfEstablisehdPeers = targetNumberOfActivePeers + 1 = 3` .
`ChurnMode` and `UseBootstrapPeers` is checked once at the start of the loop so it isn't possible run churning for active peers in one mode and churning for established peers in another.

Reply:
In this case it's fine, my argument only applies if both are equal which is allowed (although a wrong configuration), then OG will ignore the change (since it won't satisfy `sanePeerSelectionTargets`).

We should add a warning in `cardano-node` if we detect that the targets do not allow for churning if both established and active targets are equal.